python/dgl/heterograph.py

```diff
@@ -696,12 +696,8 @@ def remove_edges(self, eids, etype=None):
         for c_etype in self.canonical_etypes:
             # the target edge type
             if c_etype == (u_type, e_type, v_type):
-                old_eids = self.edges(form='eid', order='eid', etype=c_etype)
-                # trick here, eid_0 is 0 and should be handled
-                old_eids[0] += 1
-                old_eids = F.scatter_row(old_eids, eids, F.full_1d(
-                    len(eids), 0, F.dtype(old_eids), F.context(old_eids)))
-                edges[c_etype] = F.tensor(F.nonzero_1d(old_eids), dtype=F.dtype(old_eids))
+                origin_eids = self.edges(form='eid', order='eid', etype=c_etype)
+                edges[c_etype] = utils.compensate(eids, origin_eids)
```

Comment:
We don't need to create this range array right? Just an integer is enough.

Reply:
eids is supposed to be consecutive but nids are not.
The compensate can handle none consecutive origin_ids.
I think this is not a critical path, so it is ok.